src/backends/beamcoder.ts

```diff
@@ -163,6 +161,7 @@ export class BeamcoderExtractor extends BaseExtractor implements Extractor {
     filteredFrames: beamcoder.Frame[] = [];
     #streamStopped = false;
     #flushed = false;
+    #streamIndex = -1;
```

Comment:
It would be good to explain why we need this. Pretend that you're looking at this code for the first time. What is streamindex? Why do we need streamindex?

Reply:
@animanathome it's the steam index, I'm not sure what to add?

Reply:
```suggestion
    /**
     * mp4 files can contain multiple streams of different type (audio, video, text, etc.). 
     * It could be that stream 0 is not a video stream and we will work with the first video stream in the file.
     */
    #streamIndex = -1;
```

Reply:
@animanathome  maybe this?